tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/Debugger/LineProbeResolverTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="LineProbeResolverTest.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Linq;
+using Datadog.Trace.Debugger;
+using Datadog.Trace.Debugger.Configurations.Models;
+using Datadog.Trace.Debugger.Models;
+using FluentAssertions;
+using Samples.Probes.SmokeTests;
+using Xunit;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.Debugger;
+
+public class LineProbeResolverTest
+{
+    private readonly LineProbeResolver _lineProbeResolver;
+    private readonly SnapshotProbe _probeDefinition;
+
+    public LineProbeResolverTest()
+    {
+        var guidGenerator = new DeterministicGuidGenerator();
+        _lineProbeResolver = LineProbeResolver.Create();
+        var probeDefinitions = DebuggerTestHelper.GetAllProbes(typeof(LambdaSingleLine), "net461", unlisted: true, guidGenerator);
+        _probeDefinition = probeDefinitions.First().Probe;
+    }
+
+    [Fact]
+    public void ReturnsCorrectMetadata()
+    {
+        var result = _lineProbeResolver.TryResolveLineProbe(_probeDefinition, out var loc);
+        result.Status.Should().Be(LiveProbeResolveStatus.Bound);
+        var module = typeof(LambdaSingleLine).Assembly.ManifestModule;
+        var method = module.ResolveMethod(loc.MethodToken);
+        loc.MVID.Should().Be(module.ModuleVersionId);
+        method.Name.Should().Be("Run");
+    }
+
+    [Fact]
+    public void TestForwardSlashAndBackslashAreInterchangeable()
+    {
+        var result1 = _lineProbeResolver.TryResolveLineProbe(_probeDefinition, out var loc1);
+
+        // Flip the slashes
+        _probeDefinition.Where.SourceFile = _probeDefinition.Where.SourceFile.Contains(@"/") ? _probeDefinition.Where.SourceFile.Replace(@"/", @"\") : _probeDefinition.Where.SourceFile.Replace(@"\", @"/");
+
+        var result2 = _lineProbeResolver.TryResolveLineProbe(_probeDefinition, out var loc2);
+
+        result1.Status.Should().Be(LiveProbeResolveStatus.Bound);
+        result2.Status.Should().Be(LiveProbeResolveStatus.Bound);
+        (loc1.MethodToken, loc1.BytecodeOffset).Should().Be((loc2.MethodToken, loc2.BytecodeOffset));
+    }
```

Comment:
May not be necessary, and from the src I don't see why it wouldn't work, but is it worth testing the case where _some_ of the locations have the path flipped? e.g. c:\test\some/path?

Reply:
I'm purposefully not supporting the case where the **paths we store in the trie** contain a mix of both forward and back slashes. This should be fine because the inserted paths originate from the PDB, meaning they are generated by the compiler, not by humans, so we can assume that they are all consistent, so I don't think it'd be worth the extra complexity. I've added a comment to clarify this.
            
The **path we use to query the trie** however, might be typed in by a human (through the debugger's "Create New Probe" modal), and so I've added a test that that supports a mix of back- and forward- slashes in 66bc96634bc82ddf526206694ad7f7a5529a1d66.